src/music.py

```diff
@@ -121,129 +169,138 @@ def voice(self, voice):
 
-class UnvoicedStaff(_Staff):
-    '''A staff without distinct voicing (Piano pieces)'''
-    '''Notes are not attached to voices'''
-    pass
 
 # TODO: Bind measure length by global time signature
 class Measure(_Music, Collection):
     '''Collection of notes'''
 
     def __init__(self, notes=None):
-        #self._notes = notes if notes else dict()
-        self._notes = NoteContainer(notes)
-
-
-        #global _time_signature
-        #if not _time_signature: raise Exception('Time signature undefined')
-        #self._time_signature = _time_signature
-
-        #global _key_signature
-        #if not _key_signature: raise Exception('Key signature undefined')
-        #self._key_signature = _key_signature
-
-    @property
-    def key_signature(self):
-        return self._key_signature
-
-    @key_signature.setter
-    def key_signature(self, key_signature):
-        self._key_signature = key_signature
-
-    @property
-    def time_signature(self):
-        return self._time_signature
-
-    @time_signature.setter
-    def time_signature(self, time_signature):
-        self._time_signature = time_signature
+        self._notes = notes or dict()
+        self._next_count = 0.0
 
     def __setitem__(self, start, item):
-        '''To add a quarter note at beat 2: m[2] = Note('C#', 1)'''
         self._notes[start] = item
-        self._verify_duration()
-
-    def _verify_duration(self):
-        raise NotImplementedError('TODO. Verify recently-added note does not invalidate measure length wrt Time signature')
+        self._next_count = max(self._next_count, start + item.duration)
 
+    def append(self, item):
+        self._notes[self._next_count] = item
 
+        self._next_count += item.duration
 
     def __getitem__(self, start):
         return self._notes[start]
 
     def __delitem__(self, start):
-        del self.notes[start]
+        del self._notes[start]
 
     def __iter__(self, start):
-        raise NotImplementedError('TODO')
         return iter(self._notes)
 
     def __len__(self):
         '''Returns the total duration of the measure'''
-        raise NotImplementedError('TODO')
+        return self._next_count
 
     def __contains__(self, note):
         '''Returns True if note is in this measure'''
-        return note in [note for chord in measure for note in chord]
+        all_notes = []
+        for item in self._notes:
+            if type(item) == Chord:
+                all_notes.extend(item.notes)
+            elif type(item) == Note:
+                all_notes.append(item)
+
+        return note in all_notes
 
-    def append(self, note):
-        raise NotImplementedError('TODO')
 
     def extend(self, notes):
-        raise NotImplementedError('TODO.')
+        for note in notes:
+            self.append(note)
+
 
-    def add_note(self, note):
+class Chord(_Music):
+    def __init__(self, notes=None):
+        self._notes = notes or []
+        self._mingus_notes = MingusNoteContainer()
```

Comment:
The Mingus functionality is separated by keeping a MingusNoteContainer *alongside* our Chord's notes. Mingus functionalities will use the MingusNoteContainer. Mutations to the Chord should apply to both the Chord's notes and the MingusNotes